ls1intum/Artemis
# Use Text Blocks

### Checklist
- [x] This PR only changes test code
- [x] I checked that all Java tests still work on GitHub, Bamboo and locally.

### Motivation and Context
We switched to Java 15 which offers **text blocks**, see https://openjdk.java.net/jeps/378.

Replace formatted, multiline strings that are hard to read by formatted text blocks.

**Note: this excludes SQL/HQL strings which will be addressed in smaller follow-up PRs.**

### Description

Replaced the strings by text blocks. Currently, only strings that span multiple lines are changed.

Most of the changes affect only test code, with a large proportion of that being mock requests and alike.

This means that
```java
"ABC\n" + "    xyz\n"
"something long\n"
```
is replaced by
```java
"""
ABC
    xyz
something long
"""
```

#### Some things you should know about text blocks

- The position of the first `"""` introducing the text block has no effect, but it must be directly followed by a line break.
- Line breaks in the text block are preserved by default and always of the LF (`\n`) type. The one after the `"""` does not count.
- A line break can be prevented by ending the line with `\`.
- Windows line endings can be archived by ending a line with `\r` (which is then followed by the implicit `\n`).
- Whitespace at the end of the line is removed (unless an escaped space `\s` is added at the very end).
- Single `"` and double `""` do not need to be escaped in a text block.
- A text block ends with `"""`. Examples:
  ```java
  "abc" == """
           abc"""
  
  "abc\n" == """
             abc
             """
  
  "  abc\n" == """
               abc
             """
  ```
- Whitespace at the start of each line is removed as much as it is possible to do consistently. This means that the minimal number of whitespace characters that any line starts with is then removed from all lines. (See e.g. the last example of the point above)

### Steps for Testing
1. Run the server tests on your system
2. Check the code
3. Check that most of the system still works on a test server / locally.
